.github/workflows/build-and-test.yml

```diff
@@ -125,3 +132,207 @@ jobs:
         with:
           name: collector-binaries
           path: ./bin
+  loadtestpre:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.splitloadtest.outputs.matrix }}
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2.1.3
+        with:
+          go-version: 1.15
+      - name: Setup env
+        run: |
+          echo "GOPATH=$(go env GOPATH)" >> $GITHUB_ENV
+          echo "$(go env GOPATH)/bin" >> $GITHUB_PATH
+      - name: Restore module cache
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-go-modules
+        with:
+          path: /home/runner/go/pkg/mod
+          key: go-pkg-mod-${{ runner.os }}-${{ hashFiles('./go.mod') }}
+      - name: split loadtest jobs
+        id: splitloadtest
+        run: |
+          TESTS="$(make -s testbed-list-loadtest | xargs echo|sed 's/ /|/g')"
+          TESTS=(${TESTS//|/ })
+          MATRIX="{\"include\":["
+          curr=""
+          for i in "${!TESTS[@]}"; do
+            if (( i > 0 && i % 2 == 0 )); then
+              curr+="|${TESTS[$i]}"
+            else
+              if [ -n "$curr" ] && (( i>1 )); then
+                MATRIX+=",{\"test\":\"$curr\"}"
+              elif [ -n "$curr" ]; then
+                MATRIX+="{\"test\":\"$curr\"}"
+              fi
+              curr="${TESTS[$i]}"
+            fi
+          done
+          MATRIX+="]}"
+          echo "::set-output name=matrix::$MATRIX"
+  loadtest:
+    runs-on: ubuntu-latest
+    needs: [setup-environment, loadtestpre]
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.loadtestpre.outputs.matrix) }}
+    env:
+      TEST_RESULTS: testbed/tests/results/junit/results.xml
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2.1.3
+        with:
+          go-version: 1.15
+      - name: Download tool binaries
+        uses: actions/download-artifact@v2
+        with:
+          name: tool-binaries
+          path: /home/runner/go/bin
+      - name: Add execute permissions to tool binaries
+        run: chmod -R +x /home/runner/go/bin
+      - name: Setup env
+        run: |
+          echo "GOPATH=$(go env GOPATH)" >> $GITHUB_ENV
+          echo "$(go env GOPATH)/bin" >> $GITHUB_PATH
+      - name: Restore module cache
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-go-modules
+        with:
+          path: /home/runner/go/pkg/mod
+          key: go-pkg-mod-${{ runner.os }}-${{ hashFiles('./go.mod') }}
```

Comment:
This code seem to be duplicated a lot between jobs, is there a way to avoid this and have a common step that all jobs will run?

Reply:
Unfortunately I dont think there is a way to reuse steps across jobs as we can't use YAML anchors. It seems like the GitHub team is [aware](https://github.community/t/reusing-sharing-inheriting-steps-between-jobs-declarations/16851/14) this is a pain point though and working on an alternative feature soon :/